### DISCUSSION TITLE
Making PySR approximate a specific type of equation (Linear, non-linear, polynomial, etc.)

### DISCUSSION THREAD

**Discussion by Angelld23** [MilesCranmer/PySR#654] [Q&A]
I have been working in a certain material science problem and have used a few ML models such as AutoML and simple models like multiple-linear regression (MLR). I find it odd that PySR cannot come even close to the performance of even a simple MLR. I even tuned the parameter space to look over a lot and few operators, setting constrains, and increasing number of iterations and population. @MilesCranmer do you have any idea of why this might be happening? Do you have any recommendations? Although the MLR equation is not the definitive model with the best performance, it still outperformed every symbolic equation PySR gave. PySR is trying too hard to find complex symbolic equations that might fit the data. It may be interesting to be able to set the model to a starting point. For example, tell the algorithm first to use a linear regression using all the variables and start exploring from there. Another thing I need to try is to use the equation of the MLR as the input for the PySRegressor. The MLR is already very good, so my hope is that PySR might be able to improve it. I will give updates on it on the following days.

Variable handling
Also, I would like to specify that PySR use all the variables I gave in all the expressions it explores. Based on my previous knowledge of my problem, I know the variables I gave have to be fixed in the equation, so exploring symbolic expressions while missing one would be unrealistic. That could save PySR computation and time exploring equations with missing variables.

**Comment by MilesCranmer** [comment]
Can you share a bit more about the example, maybe with some code and datasets? More diagnostics would be helpful too.

Also, do you know what the equivalent complexity of the linear regression model would be, if PySR were to find it? e.g., `y = a * x + b * z` would have a complexity of 7. The default max complexity is 20 (`maxsize` parameter – which you can increase).

Also see related thread here: https://github.com/MilesCranmer/PySR/issues/536

**Reply by pukpr** [reply]
_"For example, tell the algorithm first to use a linear regression using all the variables and start exploring from there"._

I would think it would already do something like that -- for example,  the use of MLR to do Fourier series fits, a la,  s1  sin(k1 x) + c1  cos(k1 x) ... so that the s1, c1, etc values can be automatically calculated while the k1 values are evolved by the PySR algorithm.  This would require  a special case pattern match for the software to pick up these situations where the MLR requirements would be met (i..e. a 1-layer nesting of combination of factors).

**Comment by MilesCranmer** [comment]
Other related questions you should check out –
- #572 
- #273
- #161
